### ISSUE TITLE
Improve semantics of code blocks

### ISSUE THREAD

**Issue by clhenrick** [clhenrick/clhenrick.io#71]
When applying `tabindex="0"` to an element that typically isn't focusable, some browsers may compute its accessible name from its contents. 

As an improvement, an accessible name could be given via the `aria-label` or `aria-labelledby` attribute. Additionally, the code block code be conveyed as a read-only, multi-line, text area input to convey its semantics more clearly to users of assistive tech.

Example code:

```html
<pre role="textbox" aria-readonly="true" aria-multiline="true" tabindex="0" aria-label="code block of ..." rel="HTML">                         
     <code class="language-html">
	...
	...
     </code>
</pre>
```

References:
- [PrimsJS bug](https://github.com/PrismJS/prism/issues/2695)
- [Reference to related A11Y Slack discussion thread](https://web-a11y.slack.com/archives/C7KTV5CBA/p1647468193290449)
